python/pyspark/sql/conversion.py

```diff
@@ -0,0 +1,544 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import array
+import datetime
+import decimal
+from typing import Any, Callable, List, Sequence
+
+import pyarrow as pa
```

Comment:
Users might not have pyarrow installed as it's not a required package. We need to throw a user friendly error message similar to check_dependencies when pyarrow is not installed.

Reply:
added check by calling require_minimum_pyarrow_version()